pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -0,0 +1,348 @@
+package v1beta1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	osconfigv1 "github.com/openshift/api/config/v1"
+	osclientset "github.com/openshift/client-go/config/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/klog"
+	"k8s.io/utils/pointer"
+	aws "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsprovider/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	yaml "sigs.k8s.io/yaml"
+)
+
+var (
+	defaultIAMInstanceProfile = func(clusterID string) *string {
+		return pointer.StringPtr(fmt.Sprintf("%s-worker-profile", clusterID))
+	}
+	defaultSecurityGroup = func(clusterID string) string {
+		return fmt.Sprintf("%s-worker-sg", clusterID)
+	}
+	defaultSubnet = func(clusterID, az string) string {
+		return fmt.Sprintf("%s-private-%s", clusterID, az)
+	}
+)
+
+const (
+	defaultUserDataSecret    = "worker-user-data"
+	defaultCredentialsSecret = "aws-cloud-credentials"
+	defaultInstanceType      = "m4.large"
+)
```

Comment:
The ones of these that are specific to AWS should have AWS in the name, otherwise they will clash as we add more providers

It might be good to split each provider into a separate file as well, WDYT?

Reply:
++ to adding "AWS" in the name somewhere. as for breaking out to separate files i don't know, i guess it depends how large these things will get. on one hand, it seems nice to have them all central in one file in case you need to make a bunch of changes, otoh it's seems cleaner to separate them.

Reply:
thanks! Added AWS prefix and leave it in the same file in this PR. Joel feel free to decide if keep growing the file or split it based on how you feel when adding a new provider in the upcoming PR.